### ISSUE TITLE
Performance issues with PHP 7.2 and Xdebug

### ISSUE THREAD

**Issue by nplowman** [docksal/docksal#799]
**Description**

Performance is particularly slow when using PHP 7.2 with Xdebug enabled.
While I'm aware that xdebug does have some performance impact, this is particularly severe.
For example, clearing the Drupal cache on the Drupal 8 site we are working with is about 5-6 seconds with xdebug disabled, and takes almost a full minute with xdebug enabled.

Other actions that have been particularly slow are rebuilding a Patternlab instance (1.2 seconds with xdebug disabled, and 30 seconds or longer with it enabled), and viewing the Modules admin page in Drupal (5-6 seconds with it disabled, and almost a full minute with it enabled).

Other pages loads are impacted, but not nearly to this degree. These particular actions are all things which work heavily with the file system (i.e. parsing json files, yml files, etc), so that could be the common thread. But that's just a hunch.

**Steps to reproduce the issue:**

1. Spin up a Drupal 8 site using Docksal. I'm using the default settings for the 'acquia' stack, with docksal/cli:2.4-php7.2 as the CLI image.
2. Enable xdebug by adding XDEBUG_ENABLED=1 to the .docksal.env file.
3. Try running 'fin drush cr' or visiting the modules page (/admin/modules)

**Describe the results you received:**
Actions are taking unreasonably long to complete. (i.e. 30+ seconds)

**Describe the results you expected:**
Actions should take 3-5 seconds.

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

```
---------------------
COMPOSE_PROJECT_NAME_SAFE: ech_d8
COMPOSE_FILE:
/Users/nathanplowman/.docksal/stacks/volumes-bind.yml
/Users/nathanplowman/.docksal/stacks/stack-acquia.yml
/Users/nathanplowman/projects/ech_d8/.docksal/docksal.yml
ENV_FILE:
/Users/nathanplowman/projects/ech_d8/.docksal/docksal.env

PROJECT_ROOT: /Users/nathanplowman/projects/ech_d8
DOCROOT: docroot
VIRTUAL_HOST: ech-d8.docksal
VIRTUAL_HOST_ALIASES: *.ech-d8.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:32771

Docker Compose configuration
---------------------
services:
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      DOCROOT: docroot
      HOST_GID: '20'
      HOST_UID: '501'
      SECRET_ACAPI_EMAIL: null
      SECRET_ACAPI_KEY: null
      SECRET_BLACKFIRE_CLIENT_ID: null
      SECRET_BLACKFIRE_CLIENT_TOKEN: null
      SECRET_PLATFORMSH_CLI_TOKEN: null
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      VIRTUAL_HOST: ech-d8.docksal
      XDEBUG_CONFIG: remote_connect_back=0 remote_host=192.168.64.1
      XDEBUG_ENABLED: '0'
    hostname: cli
    image: docksal/cli:2.4-php7.2
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - cli_home:/home/docker:rw
    - project_root:/var/www:rw,nocopy
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: null
      MYSQL_DATABASE: default
      MYSQL_INITDB_SKIP_TZINFO: null
      MYSQL_ONETIME_PASSWORD: null
      MYSQL_PASSWORD: user
      MYSQL_RANDOM_ROOT_PASSWORD: null
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.2-mysql-5.6
    ports:
    - 3306/tcp
    volumes:
    - db_data:/var/lib/mysql:rw
    - project_root:/var/www:ro,nocopy
  mail:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: mail
    image: mailhog/mailhog
    labels:
      io.docksal.cert-name: none
      io.docksal.virtual-host: mail.ech-d8.docksal,mail.ech-d8.docksal.*
      io.docksal.virtual-port: '8025'
    volumes:
    - project_root:/var/www:ro,nocopy
  memcached:
    command:
    - -m
    - '128'
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: memcached
    image: memcached:1.4-alpine
  solr:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: solr
    image: docksal/solr:1.0-solr4
    labels:
      io.docksal.cert-name: none
      io.docksal.virtual-host: solr.ech-d8.docksal,solr.ech-d8.docksal.*
      io.docksal.virtual-port: '8983'
    volumes:
    - project_root:/var/www:ro,nocopy
  varnish:
    depends_on:
      web:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      VARNISH_BACKEND_HOST: web
    hostname: varnish
    image: docksal/varnish:1.1-varnish5
    labels:
      io.docksal.cert-name: none
      io.docksal.virtual-host: varnish.ech-d8.docksal,varnish.ech-d8.docksal.*
    volumes:
    - project_root:/var/www:ro,nocopy
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.cert-name: none
      io.docksal.project-root: /Users/nathanplowman/projects/ech_d8
      io.docksal.virtual-host: ech-d8.docksal,*.ech-d8.docksal,ech-d8.docksal.*
    volumes:
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  cli_home: {}
  db_data: {}
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /Users/nathanplowman/projects/ech_d8
      o: bind
      type: none

---------------------

```

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
███  OS
Darwin Mac OS X 10.13.6
Darwin Nathans-MacBook-Pro-7.local 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64

███  ENVIRONMENT
MODE : VirtualBox VM
DOCKER_HOST : tcp://192.168.64.100:2376
DOCKSAL_NFS_PATH : /Users/nathanplowman/projects
NFS EXPORTS:
# VAGRANT-BEGIN: 501 ed886626-fcf0-4202-b4e6-69f551b51623
#"/Users/nathanplowman/Sites/tek.com" 192.168.56.132 -alldirs -mapall=501:20
# VAGRANT-END: 501 ed886626-fcf0-4202-b4e6-69f551b51623







# VAGRANT-BEGIN: 501 959250a2-c118-4b36-a7e7-4124bcb89482
"/Users/nathanplowman/Laravel" 192.168.10.10 -alldirs -mapall=501:20
# VAGRANT-END: 501 959250a2-c118-4b36-a7e7-4124bcb89482
# <ds-nfs docksal
/Users/nathanplowman/projects 127.0.0.1 192.168.64.100 -alldirs -maproot=0:0
# ds-nfs>

███  FIN
fin version: 1.69.1

███  DOCKER COMPOSE
EXPECTED VERSION: 1.21.1
docker-compose version 1.21.1, build 5a3f1a3
docker-py version: 3.3.0
CPython version: 3.6.4
OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018

███  DOCKER
EXPECTED VERSION: 18.03.1-ce

Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:13:02 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:23:03 2018
  OS/Arch:      linux/amd64
  Experimental: false

███  DOCKER MACHINE
EXPECTED VERSION: 0.14.0
docker-machine version 0.14.0, build 89b8332

NAME      ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
docksal   *        virtualbox   Running   tcp://192.168.64.100:2376           v18.03.1-ce   

███  DOCKSAL: PROJECTS
project             STATUS              virtual host                                            project root
ech_d8              Up 45 seconds       ech-d8.docksal,*.ech-d8.docksal,ech-d8.docksal.*        /Users/nathanplowman/projects/ech_d8
ech                 Up 5 days           ech.docksal,*.ech.docksal,echd.docksal,*.echd.docksal   /Users/nathanplowman/projects/ech

███  DOCKSAL: VIRTUAL HOSTS
http://*.ech-d8.docksal
http://*.ech.docksal
http://*.echd.docksal
http://ech-d8.docksal.*
http://ech-d8.docksal
http://ech.docksal
http://echd.docksal
http://mail.ech-d8.docksal.*
http://mail.ech-d8.docksal
http://solr.ech-d8.docksal.*
http://solr.ech-d8.docksal
http://solr.ech.docksal.*
http://solr.ech.docksal
http://varnish.ech-d8.docksal.*
http://varnish.ech-d8.docksal
http://varnish.ech.docksal.*
http://varnish.ech.docksal
http://webmail.ech.docksal

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID        IMAGE                          COMMAND                  CREATED             STATUS                    PORTS                                                    NAMES
1a34b128977e        docksal/varnish:1.1-varnish5   "/opt/startup.sh"        35 seconds ago      Up 35 seconds             80/tcp, 6082/tcp                                         ech_d8_varnish_1
0d3a016ac4d9        docksal/web:2.1-apache2.4      "httpd-foreground"       46 seconds ago      Up 46 seconds             80/tcp, 443/tcp                                          ech_d8_web_1
a89f4af369de        docksal/cli:2.4-php7.2         "/opt/startup.sh sup…"   47 seconds ago      Up 47 seconds (healthy)   22/tcp, 3000/tcp, 9000/tcp                               ech_d8_cli_1
c1e93d517d47        mailhog/mailhog                "MailHog"                7 days ago          Up 2 days                 1025/tcp, 8025/tcp                                       ech_d8_mail_1
33baffa08480        docksal/varnish:1.1-varnish5   "/opt/startup.sh"        8 days ago          Up 5 days                 80/tcp, 6082/tcp                                         ech_varnish_1
16b5710ffd19        docksal/web:2.1-apache2.4      "httpd-foreground"       8 days ago          Up 5 days                 80/tcp, 443/tcp                                          ech_web_1
e0af90f27e31        docksal/cli:2.1-php7.1         "/opt/startup.sh sup…"   8 days ago          Up 5 days (healthy)       22/tcp, 3000/tcp, 9000/tcp                               ech_cli_1
278719ba5698        docksal/solr:1.0-solr3         "java -jar start.jar"    4 weeks ago         Up 5 days                 8983/tcp                                                 ech_solr_1
81f3619b278c        docksal/db:1.2-mysql-5.6       "/entrypoint.sh mysq…"   4 weeks ago         Up 5 days                 0.0.0.0:32770->3306/tcp, 0.0.0.0:33062->3306/tcp         ech_db_echd_1
03d48caa19bb        docksal/db:1.2-mysql-5.6       "/entrypoint.sh mysq…"   4 weeks ago         Up 5 days                 0.0.0.0:32769->3306/tcp, 0.0.0.0:33061->3306/tcp         ech_db_ech_1
7cc9939fb9f7        memcached:1.4-alpine           "docker-entrypoint.s…"   4 weeks ago         Up 5 days                 11211/tcp                                                ech_memcached_1
303acda94ec0        docksal/db:1.2-mysql-5.6       "/entrypoint.sh mysq…"   4 weeks ago         Up 20 hours               0.0.0.0:32772->3306/tcp                                  ech_db_1
cab894999c9a        docksal/solr:1.0-solr4         "/opt/solr/bin/solr …"   4 weeks ago         Up 2 days                 8983/tcp                                                 ech_d8_solr_1
0ed7792a79b7        docksal/db:1.2-mysql-5.6       "/entrypoint.sh mysq…"   4 weeks ago         Up 2 days                 0.0.0.0:32771->3306/tcp                                  ech_d8_db_1
b1b34445027d        memcached:1.4-alpine           "docker-entrypoint.s…"   4 weeks ago         Up 2 days                 11211/tcp                                                ech_d8_memcached_1
7d958c90610a        docksal/ssh-agent:1.0          "/run.sh ssh-agent"      4 weeks ago         Up 5 days                                                                          docksal-ssh-agent
b1657ccff0fe        docksal/dns:1.0                "/opt/entrypoint.sh …"   4 weeks ago         Up 5 days                 192.168.64.100:53->53/udp                                docksal-dns
c482e5c4367e        docksal/vhost-proxy:1.3        "docker-entrypoint.s…"   4 weeks ago         Up 5 days                 192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
22c3698d0253        mailhog/mailhog                "MailHog"                3 months ago        Up 5 days                 80/tcp, 1025/tcp, 8025/tcp                               ech_mail_1

███  DOCKER: NETWORKS
NETWORK ID          NAME                        DRIVER              SCOPE
ba738f617265        _default                    bridge              local
d694309abed9        bridge                      bridge              local
add7c9d4676d        ech_d8_default              bridge              local
9e4f6eb91a45        ech_default                 bridge              local
11cdafcd5bcd        ffwagency_default           bridge              local
61fae5d444b4        headlessdrupal_default      bridge              local
32cb687e2b40        host                        host                local
1ac67804b0bf        juilliard_default           bridge              local
f5512dcfe7f5        juilliard_journal_default   bridge              local
cf7d8f220755        none                        null                local
c9c58202846e        tekcom_default              bridge              local
29b4ce0a225e        tianjin_default             bridge              local

███  VIRTUALBOX
EXPECTED VERSION: 5.2.2
5.2.2r119230

███  DOCKSAL MOUNTS
192.168.64.1:/Users/nathanplowman/projects/ech on /mnt/sda1/var/lib/docker/volumes/ech_project_root/_data type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)
192.168.64.1:/Users/nathanplowman/projects/ech_d8 on /mnt/sda1/var/lib/docker/volumes/ech_d8_project_root/_data type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)
192.168.64.1:/Users/nathanplowman/projects on /Users/nathanplowman/projects type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)

███  HDD Usage
Filesystem                Size      Used Available Use% Mounted on
/dev/sda1                93.6G     55.0G     33.7G  62% /mnt/sda1
  ```

</details>

**Comment by lmakarov** [comment]
@nplowman that's for the report.

Quick note - the default PHP version for all stacks is 7.1 at the moment

https://github.com/docksal/docksal/blob/0056960699179ceff27b621623907922a084bf31/stacks/services.yml#L87

You must be setting an override in either `docksal.yml` or `docksal.env` in your project.

**Comment by achekulaev** [comment]
Please update to the latest version of Docksal (released yesterday) and tell if it solves the issues for you.

**Comment by christianwiedemann** [comment]
Hi,
got the same issue with php7.2 and php 7.3 (docksal/cli:edge-php7.3) with xdebug enabled:
(If I should open an new issue let me know. )

**Some stats from my linux host**: 

2.5-php7.1: drush cr with xdebug: 11.43 sec
2.5-php7.1: drush cr without xdebug: 4.71 sec

2.5-php7.2: drush cr with xdebug: 28.39 sec
2.5-php7.2: drush cr without xdebug: 4.64 sec

cli:edge-php7.3: drush cr with xdebug: 28.23 sec
cli:edge-php7.3: drush cr without xdebug: 3.79 sec

**Comment by christianwiedemann** [comment]
Seems that the slow performance comes with Xdebug 2.7.0beta1. I downgraded the php 7.2 image to use Xdebug v2.6.1. and than the performance is as expected.

**Comment by lmakarov** [comment]
@charginghawk many thanks for researching this and pinpointing the root cause.

We are installing xdebug from [source](https://github.com/xdebug/xdebug.git) (latest versions) for PHP 7.2 and 7.3.
I will check if we can stop doing that at this point or just pin the version of xdebug as you suggested.

**Comment by achekulaev** [comment]
@christianwiedemann thanks for investigation!

**Comment by lmakarov** [comment]
Resolved and released in [docksal/cli v2.6.0](https://github.com/docksal/service-cli/releases/tag/v2.6.0)
